From 99659a911616298190930b2bf370bd2794973723 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Sat, 2 Jul 2016 12:43:58 +0200
Subject: [PATCH] iproute2: update ip-link.8 for geneve tunnels

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1339178
Upstream Status: iproute2.git commit c1a1d8bc4c2f1
Conflicts: Context changes due to missing lowpan interface type support
           and previously added MACVLAN/MACVTAP support.

commit c1a1d8bc4c2f1da2f4ed5f11e3d9edc276c0c8d1
Author: John W. Linville <linville@tuxdriver.com>
Date:   Fri May 22 14:46:46 2015 -0400

    iproute2: update ip-link.8 for geneve tunnels

    Signed-off-by: John W. Linville <linville@tuxdriver.com>
---
 man/man8/ip-link.8.in | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/man/man8/ip-link.8.in b/man/man8/ip-link.8.in
index 5e4862f..21503ff 100644
--- a/man/man8/ip-link.8.in
+++ b/man/man8/ip-link.8.in
@@ -60,7 +60,8 @@ ip-link \- network device configuration
 .BR ip6gretap " |"
 .BR vti " |"
 .BR nlmon " |"
-.BR ipvlan " ]"
+.BR ipvlan " |"
+.BR geneve " ]"
 
 .ti -8
 .BI "ip link delete " DEVICE
@@ -231,6 +232,9 @@ Link types:
 .sp
 .BR ipvlan
 - Interface for L3 (IPv6/IPv4) based VLANs
+.sp
+.BR geneve
+- GEneric NEtwork Virtualization Encapsulation
 .in -8
 
 .TP
@@ -659,6 +663,26 @@ the following additional arguments are supported:
 - specifies the mode (datagram or connected) to use.
 
 .TP
+GENEVE Type Support
+For a link of type
+.I GENEVE
+the following additional arguments are supported:
+
+.BI "ip link add " DEVICE
+.BI type " geneve " id " ID " remote " IPADDR"
+
+.in +8
+.sp
+.BI  id " VNI "
+- specifies the Virtual Network Identifer to use.
+
+.sp
+.BI remote " IPADDR"
+- specifies the unicast destination IP address to use in outgoing packets.
+
+.in -8
+
+.TP
 MACVLAN and MACVTAP Type Support
 For a link of type
 .I MACVLAN
-- 
1.8.3.1

